### ISSUE TITLE
Saving the Navigator in the Metadata

### ISSUE THREAD

**Issue by CSSFrancis** [hyperspy/hyperspy#3104]
**Edited: I realized that my bug was related to a different issue.**

#### Describe the functionality you would like to see.
It would be nice to be able to save n dimensional navigators instead of just 2 dimensional navigators.

In my case for large datasets with 3 navigation dimensions, (x,y,time) it takes a long time to compute the navigator.  It would also be nice to have the navigator update as you cycle through the time axis. 

#### Additional information

It shouldn't be too hard of a change but it would go a long way towards making visualizing large datasets easier.  Another thing to consider would be to up the caching of the active chunk and surrounding chunks.  That way while it might take a little longer to first plot you could quickly explore the dataset without experiencing slow downs.

**Comment by ericpre** [comment]
Yes, indeed it makes sense. I had a quick look and it would need some changes to add support for navigator with `navigation_dimension` > 2.
The navigator indices will need to be updated when changing the signal indices using events, something along the line of:
```python
def link_navigator_indices(*args, **kargs):
    navigator.axes_manager.indices = s.axes_manager.indices[:-2]

s.axes_manager.events.indices_changed.connect(link_navigator_indices, [])
```
